src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMFileSystem.java

```diff
@@ -328,10 +318,17 @@ public SCMFileSystem build(@NonNull SCMSource source, @NonNull SCMHead head, @Ch
                         rev = new AbstractGitSCMSource.SCMRevisionImpl(head, ref.getObject().getSha());
                     }
                 }
-                return new GitHubSCMFileSystem(github, repo, refName, rev);
+
+                // Instead of calling release in many case and skipping this one case
+                // Make another call to connect() for this case
+                // and always release the existing instance as part of finally block.
+                // The result is the same but with far fewer code paths calling release().
```

Comment:
There is no performance issue here, right, we just look up the same connection that would have been created or obtained from the cache at the beginning of this method? And we just need to call `connect` here so that the usages goes up by 1 to account for the fact that we will always decrease the number of usages by 1 in the finally block?

Reply:
Yes. We're not creating a new instance, just incrementing a reference count.